source/tutorials/kubernetes.rst

```diff
@@ -3,170 +3,340 @@
 
-#. Enable IP forwarding to avoid kubeadm `preflight check`_ errors:
 
-   Create (or edit if it exists) the file :file:`/etc/sysctl.d/60-k8s.conf`
-   and include the following line:
+#. Enable IP forwarding:
 
-   .. code-block:: bash
+   - Create the file :file:`/etc/sysctl.d/60-k8s.conf` to set the
+     :command:`net.ipv4.ip_forward` parameter
+
+     .. code-block:: bash
+
+        sudo mkdir -p /etc/sysctl.d/
+
+        sudo tee /etc/sysctl.d/99-kubernetes-cri.conf > /dev/null <<EOF
+        net.bridge.bridge-nf-call-iptables  = 1
+        net.ipv4.ip_forward                 = 1
+        net.bridge.bridge-nf-call-ip6tables = 1
+        EOF
+
+   - Apply the change:
+
+     .. code-block:: bash
+
+        sudo sysctl --system
 
-      net.ipv4.ip_forward = 1
 
-   Apply the change:
+#. Disable swap:
 
    .. code-block:: bash
 
-      sudo systemctl restart systemd-sysctl
+      sudo systemctl mask $(sed -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps) 2>/dev/null
+      sudo swapoff -a
 
-#. Enable the kubelet service:
+   .. warning::
+
+      Kubernetes is designed to work without swap. Performance degradation of other workloads can occur
+      with swap disabled on systems with constrained memory resources.
+
+#. Add the the system's hostname to the :file:`/etc/hosts` file. Kubernetes
+   will read this file to locate the master host.
+
+    .. code-block:: bash
+
+       echo "127.0.0.1 localhost `hostname`" | sudo tee --append /etc/hosts
+
+
+#. Enable the kubelet agent service to start at boot automatically:
 
    .. code-block:: bash
 
       sudo systemctl enable kubelet.service
 
-#. Disable swap using one of the following methods, either:
 
-   a) Temporarily:
+Initialize the master node
+**************************
+
+In Kubernetes, a master node is part of the `Kubernetes Control Plane
+<https://kubernetes.io/docs/concepts/#kubernetes-control-plane>`_. 
 
-      .. code-block:: bash
+Initializing a new Kubernetes cluster involves crafting a :command:`kubeadm
+init` command. Adding parameters to this command can control the fundamental
+operating components of the cluster. This means it is important to understand
+and choose network and runtime options before running a :command:`kubeadm
+init` command.
 
-         sudo swapoff -a
 
-      .. note::
+Choose a pod network add-on
+===========================
 
-         Swap will be enabled at next reboot, causing failures in
-         your cluster if you choose this method.
+In Kubernetes, a `pod
+<https://kubernetes.io/docs/concepts/workloads/pods/pod/>`_ is a group of one
+or more containers and is the smallest deployable unit of computing in a
+Kubernetes cluster. Pods have shared storage/network internally but
+communication between pods requires additional configuration. If you want your
+pods to be able to communicate with each other you must choose and install a
+`pod network add-on`_. Otherwise, this section can be skipped. 
```

Comment:
Is there any case where someone would want a kubernetes cluster but not have a pod network add on?  If not then I would drop the "this section can be skipped" part.